<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>610</width>
    <height>582</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>PeakEngine Editor</string>
  </property>
  <property name="windowIcon">
   <iconset resource="../data/editor.qrc">
    <normaloff>:/icons/tool-animator.png</normaloff>:/icons/tool-animator.png</iconset>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QHBoxLayout" name="horizontalLayout_2">
    <property name="spacing">
     <number>0</number>
    </property>
    <property name="margin">
     <number>0</number>
    </property>
    <item>
     <widget class="OpenGLView" name="openglview" native="true"/>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>610</width>
     <height>27</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionNewObject"/>
    <addaction name="actionOpenObject"/>
    <addaction name="actionCloseObject"/>
    <addaction name="separator"/>
    <addaction name="actionNewProject"/>
    <addaction name="actionOpenProject"/>
    <addaction name="actionCloseProject"/>
    <addaction name="actionRecentProjects"/>
    <addaction name="separator"/>
    <addaction name="actionExit"/>
   </widget>
   <widget class="QMenu" name="menuEdit">
    <property name="title">
     <string>Edit</string>
    </property>
    <addaction name="actionConfigure"/>
   </widget>
   <widget class="QMenu" name="menuView">
    <property name="title">
     <string>View</string>
    </property>
   </widget>
   <widget class="QMenu" name="menuHelp">
    <property name="title">
     <string>Help</string>
    </property>
    <addaction name="actionAbout"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuEdit"/>
   <addaction name="menuView"/>
   <addaction name="menuHelp"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <widget class="QToolBar" name="toolBar">
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionStop"/>
   <addaction name="actionPause"/>
   <addaction name="actionPlay"/>
   <addaction name="actionRecompile"/>
   <addaction name="actionLaunch"/>
   <addaction name="separator"/>
   <addaction name="actionNewObject"/>
   <addaction name="actionOpenObject"/>
   <addaction name="actionCloseObject"/>
   <addaction name="separator"/>
   <addaction name="actionConfigure"/>
  </widget>
  <widget class="QDockWidget" name="dockProject">
   <property name="windowTitle">
    <string>Project</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>1</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents">
    <layout class="QVBoxLayout" name="verticalLayout">
     <item>
      <widget class="QTreeView" name="objectList"/>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QDockWidget" name="dockInspector">
   <property name="floating">
    <bool>false</bool>
   </property>
   <property name="windowTitle">
    <string>Inspector</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>2</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents_2">
    <layout class="QVBoxLayout" name="verticalLayout_2">
     <item>
      <widget class="QStackedWidget" name="inspectorStack">
       <property name="currentIndex">
        <number>2</number>
       </property>
       <widget class="QWidget" name="entityInspector">
        <layout class="QHBoxLayout" name="horizontalLayout_3">
         <property name="spacing">
          <number>0</number>
         </property>
         <property name="margin">
          <number>0</number>
         </property>
         <item>
          <widget class="QToolBox" name="toolBox">
           <property name="currentIndex">
            <number>0</number>
           </property>
           <property name="tabSpacing">
            <number>6</number>
           </property>
           <widget class="QWidget" name="page">
            <property name="geometry">
             <rect>
              <x>0</x>
              <y>0</y>
              <width>96</width>
              <height>92</height>
             </rect>
            </property>
            <attribute name="label">
             <string>Components</string>
            </attribute>
            <layout class="QVBoxLayout" name="verticalLayout_3">
             <property name="spacing">
              <number>0</number>
             </property>
             <property name="margin">
              <number>0</number>
             </property>
             <item>
              <widget class="QTreeView" name="treeView"/>
             </item>
            </layout>
           </widget>
           <widget class="QWidget" name="page_2">
            <property name="geometry">
             <rect>
              <x>0</x>
              <y>0</y>
              <width>96</width>
              <height>92</height>
             </rect>
            </property>
            <attribute name="label">
             <string>Properties</string>
            </attribute>
            <layout class="QVBoxLayout" name="verticalLayout_4">
             <property name="spacing">
              <number>0</number>
             </property>
             <property name="margin">
              <number>0</number>
             </property>
             <item>
              <widget class="QListView" name="listView"/>
             </item>
            </layout>
           </widget>
          </widget>
         </item>
        </layout>
       </widget>
       <widget class="QWidget" name="nothingInspector">
        <layout class="QHBoxLayout" name="horizontalLayout_4">
         <item>
          <widget class="QLabel" name="label">
           <property name="text">
            <string>Object not opened.</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
       <widget class="QWidget" name="codeComponentInspector">
        <layout class="QVBoxLayout" name="verticalLayout_6">
         <item>
          <widget class="QPushButton" name="codeCompShow">
           <property name="text">
            <string>Show source code</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QListView" name="codeCompProperties"/>
         </item>
        </layout>
       </widget>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QDockWidget" name="dockConsole">
   <property name="windowTitle">
    <string>Console</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>8</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents_3">
    <layout class="QHBoxLayout" name="horizontalLayout">
     <item>
      <widget class="QPlainTextEdit" name="console">
       <property name="readOnly">
        <bool>true</bool>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <action name="actionPlay">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="../data/editor.qrc">
     <normaloff>:/icons/media-playback-start.png</normaloff>:/icons/media-playback-start.png</iconset>
   </property>
   <property name="text">
    <string>Play</string>
   </property>
  </action>
  <action name="actionPause">
   <property name="icon">
    <iconset resource="../data/editor.qrc">
     <normaloff>:/icons/media-playback-pause.png</normaloff>:/icons/media-playback-pause.png</iconset>
   </property>
   <property name="text">
    <string>Pause</string>
   </property>
   <property name="toolTip">
    <string>Pause</string>
   </property>
  </action>
  <action name="actionStop">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="../data/editor.qrc">
     <normaloff>:/icons/media-playback-stop.png</normaloff>:/icons/media-playback-stop.png</iconset>
   </property>
   <property name="text">
    <string>Stop</string>
   </property>
   <property name="toolTip">
    <string>Stop</string>
   </property>
  </action>
  <action name="actionOpenProject">
   <property name="icon">
    <iconset resource="../data/editor.qrc">
     <normaloff>:/icons/project-open.png</normaloff>:/icons/project-open.png</iconset>
   </property>
   <property name="text">
    <string>Open project</string>
   </property>
  </action>
  <action name="actionCloseProject">
   <property name="icon">
    <iconset resource="../data/editor.qrc">
     <normaloff>:/icons/project-development-close.png</normaloff>:/icons/project-development-close.png</iconset>
   </property>
   <property name="text">
    <string>Close project</string>
   </property>
  </action>
  <action name="actionNewProject">
   <property name="icon">
    <iconset resource="../data/editor.qrc">
     <normaloff>:/icons/project-development-new-template.png</normaloff>:/icons/project-development-new-template.png</iconset>
   </property>
   <property name="text">
    <string>New project</string>
   </property>
   <property name="toolTip">
    <string>New project</string>
   </property>
  </action>
  <action name="actionExit">
   <property name="icon">
    <iconset resource="../data/editor.qrc">
     <normaloff>:/icons/application-exit.png</normaloff>:/icons/application-exit.png</iconset>
   </property>
   <property name="text">
    <string>Exit</string>
   </property>
  </action>
  <action name="actionNewObject">
   <property name="icon">
    <iconset resource="../data/editor.qrc">
     <normaloff>:/icons/document-new.png</normaloff>:/icons/document-new.png</iconset>
   </property>
   <property name="text">
    <string>New object</string>
   </property>
  </action>
  <action name="actionOpenObject">
   <property name="icon">
    <iconset resource="../data/editor.qrc">
     <normaloff>:/icons/document-open.png</normaloff>:/icons/document-open.png</iconset>
   </property>
   <property name="text">
    <string>Open object</string>
   </property>
   <property name="toolTip">
    <string>Open object</string>
   </property>
  </action>
  <action name="actionCloseObject">
   <property name="icon">
    <iconset resource="../data/editor.qrc">
     <normaloff>:/icons/document-close.png</normaloff>:/icons/document-close.png</iconset>
   </property>
   <property name="text">
    <string>Close object</string>
   </property>
   <property name="toolTip">
    <string>Close object</string>
   </property>
  </action>
  <action name="actionRecompile">
   <property name="icon">
    <iconset resource="../data/editor.qrc">
     <normaloff>:/icons/view-refresh.png</normaloff>:/icons/view-refresh.png</iconset>
   </property>
   <property name="text">
    <string>Recompile</string>
   </property>
  </action>
  <action name="actionAbout">
   <property name="text">
    <string>About</string>
   </property>
  </action>
  <action name="actionLaunch">
   <property name="icon">
    <iconset resource="../data/editor.qrc">
     <normaloff>:/icons/tool-animator.png</normaloff>:/icons/tool-animator.png</iconset>
   </property>
   <property name="text">
    <string>Launch</string>
   </property>
   <property name="toolTip">
    <string>Launch game</string>
   </property>
  </action>
  <action name="actionRecentProjects">
   <property name="icon">
    <iconset resource="../data/editor.qrc">
     <normaloff>:/icons/document-open-recent.png</normaloff>:/icons/document-open-recent.png</iconset>
   </property>
   <property name="text">
    <string>Recent projects</string>
   </property>
  </action>
  <action name="actionConfigure">
   <property name="icon">
    <iconset resource="../data/editor.qrc">
     <normaloff>:/icons/configure.png</normaloff>:/icons/configure.png</iconset>
   </property>
   <property name="text">
    <string>Configure project</string>
   </property>
   <property name="toolTip">
    <string>Configure project</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>OpenGLView</class>
   <extends>QWidget</extends>
   <header>OpenGLView.hpp</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="../data/editor.qrc"/>
 </resources>
 <connections/>
</ui>
